pkg/template/controller/templateinstance_controller.go

```diff
@@ -61,13 +68,13 @@ type TemplateInstanceController struct {
 }
 
 // NewTemplateInstanceController returns a new TemplateInstanceController.
-func NewTemplateInstanceController(config *rest.Config, oc client.Interface, kc kclientsetinternal.Interface, templateclient templateclient.Interface, informer internalversion.TemplateInstanceInformer) *TemplateInstanceController {
+func NewTemplateInstanceController(config *rest.Config, kc kclientsetinternal.Interface, buildClient buildclient.Interface, templateClient templateclient.Interface, informer internalversion.TemplateInstanceInformer) *TemplateInstanceController {
```

Comment:
Why not just passing `buildclient.BuildInterface` here? I doubt you'll be needing the `Discovery` part of that `Interface`

Reply:
@soltysh i'm not trying to fix pre-existing code :) will leave that as follow up for @jim-minter